server/tests/integration_test.rs

```diff
@@ -0,0 +1,998 @@
+    }
+}
+
+/// Send messages to the ModelarDB server with Arrow Flight through the `do_put()` endpoint.
+fn send_messages_to_arrow_flight_server(
+    rt: &Runtime,
+    mut client: &mut FlightServiceClient<Channel>,
+    flight_data_vec: Vec<FlightData>,
+) -> std::result::Result<(), ()> {
+    rt.block_on(async {
+        let flight_data_stream = stream::iter(flight_data_vec);
+
+        let mut _streaming = client.do_put(flight_data_stream).await;
+
+        Ok(())
+    })
+}
+
+/// Flush the data in the StorageEngine to disk through the `do_action()` endpoint.
+fn flush_data_to_disk(rt: &Runtime, fsc: &mut FlightServiceClient<Channel>) -> Result<()> {
+    let action = Action {
+        r#type: "Flush".to_owned(),
+        body: vec![],
+    };
+    let request = Request::new(action);
+
+    rt.block_on(async {
+        fsc.do_action(request).await?.into_inner().message().await?;
+        Ok(())
+    })
+}
+
+/// Execute a query on the ModelarDB server through the `do_get()`endpoint.
+fn execute_query(
+    rt: &Runtime,
+    fsc: &mut FlightServiceClient<Channel>,
+    query: &str,
+) -> Result<Vec<RecordBatch>> {
+    rt.block_on(async {
+        //Execute query
+        let ticket_data = query.to_owned().into_bytes();
+        let ticket = arrow_flight::Ticket {
+            ticket: ticket_data,
+        };
+        let mut stream = fsc.do_get(ticket).await?.into_inner();
+
+        //Get schema of result set
+        let flight_data = stream
+            .message()
+            .await?
+            .ok_or("transport error: no messages received")?;
+        let schema = Arc::new(Schema::try_from(&flight_data)?);
+
+        //Get data in result set
+        let mut results = vec![];
+        while let Some(flight_data) = stream.message().await? {
+            let dictionaries_by_id = HashMap::new();
+            let record_batch =
+                flight_data_to_arrow_batch(&flight_data, schema.clone(), &dictionaries_by_id)?;
+            results.push(record_batch);
+        }
+        Ok(results)
+    })
+}
+
+/// Retrieve the table names currently in the ModelarDB server and return as a Vector.
+fn retrieve_table_names(
+    rt: &Runtime,
+    fsc: &mut FlightServiceClient<Channel>,
+) -> Result<Vec<String>> {
+    let criteria = Criteria { expression: vec![] };
+    let request = Request::new(criteria);
+
+    rt.block_on(async {
+        let mut stream = fsc.list_flights(request).await?.into_inner();
+        let flights = stream
+            .message()
+            .await?
+            .ok_or("transport error: no messages received")?;
+
+        let mut table_names = vec![];
+        if let Some(fd) = flights.flight_descriptor {
+            for table in fd.path {
+                table_names.push(table);
+            }
+        }
+        Ok(table_names)
+    })
+}
+
+/// Reconstructs the record_batch based on the queried RecordBatch and the tag and schema from the generated RecordBatch.
```

Comment:
Should this be "Reconstructs the `original` record batch" instead?

Reply:
It might be necessary to add a bit more to this comment. It is a bit hard to understand the purpose of this function.

Reply:
Any suggestions? I made a few changes, but I don't really see how it can be clearer. It does what it says in the comment. Or do you mean context? Like why I need to reconstruct them?

Reply:
I think it would be good to include some context on why it is necessary to reconstruct.

Reply:
Fixed.